blobstore/src/main/java/org/jclouds/blobstore/internal/BaseBlobStore.java

```diff
@@ -275,4 +285,20 @@ public String copyBlob(String fromContainer, String fromName, String toContainer
          Closeables2.closeQuietly(is);
       }
    }
+
+   // TODO: parallel uploads
+   @Beta
+   protected String putMultipartBlob(String container, Blob blob, PutOptions overrides) {
+      MultipartUpload mpu = initiateMultipartUpload(container, blob.getMetadata());
+      List<MultipartPart> parts = Lists.newArrayList();
+      long contentLength = blob.getMetadata().getContentMetadata().getContentLength();
+      long partSize = getMaximumMultipartPartSize();  // TODO: optimal?
```

Comment:
I reparented the S3 `MultipartUploadSlicingAlgorithm` to core so we have the same algorithm as before.